idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/frontend/api/fir/components/KtFirSymbolDeclarationRendererProvider.kt

```diff
@@ -39,9 +52,109 @@ internal class KtFirSymbolDeclarationRendererProvider(
         val phaseNeeded =
             if (options.renderContainingDeclarations) FirResolvePhase.BODY_RESOLVE else FirResolvePhase.IMPLICIT_TYPES_BODY_RESOLVE
 
-        return (containingSymbol ?: symbol).firRef.withFir(phaseNeeded) { fir ->
+        return symbol.firRef.withFir(phaseNeeded) { fir ->
             val containingFir = containingSymbol?.firRef?.withFirUnsafe { it }
             FirIdeRenderer.render(fir, containingFir, options, fir.session)
         }
     }
+
+    override fun getIcon(symbol: KtSymbol): Icon? {
```

Comment:
Let's please move icon provider out of `KtAnalysisSession` as it has nothing to do with resolve and can be implemented without directly using FIR

Reply:
Thanks. Let me put it in a util object.